backend/inspirehep/records/api/literature.py

```diff
@@ -157,23 +184,36 @@ def update_refs_to_conferences(data):
                 continue
             conference["conference_record"] = get_ref_from_pid("con", pid)
 
-    def update(self, data, disable_orcid_push=False, *args, **kwargs):
+    def update(
+        self,
+        data,
+        disable_orcid_push=False,
+        disable_relations_update=False,
+        *args,
+        **kwargs,
+    ):
         with db.session.begin_nested():
             LiteratureRecord.update_authors_signature_blocks_and_uuids(data)
             LiteratureRecord.update_refs_to_conferences(data)
-
             data = self.add_files(data)
             super().update(data, *args, **kwargs)
 
-            if disable_orcid_push:
-                LOGGER.info(
-                    "Record ORCID PUSH disabled",
-                    recid=self.get("control_number"),
-                    uuid=str(self.id),
-                )
-            else:
-                push_to_orcid(self)
-            self.push_authors_phonetic_blocks_to_redis()
+            self.update_authors_records_table()
+            if not disable_relations_update:
+                self.update_refs_in_citation_table()
+                self.update_conference_paper_and_proccedings()
+                self.update_institution_relations()
+                self.update_experiment_relations()
+
+        if disable_orcid_push:
+            LOGGER.info(
+                "Record ORCID PUSH disabled",
+                recid=self.get("control_number"),
+                uuid=str(self.id),
+            )
+        else:
+            push_to_orcid(self)
+        self.push_authors_phonetic_blocks_to_redis()
```

Comment:
This is almost identical to create() except super().update/super().create calls, can you maybe split this to two separate methods which will be called before and after super().update/super().create calls? So we won't have to keep two identical peaces of code separate?

Reply:
we need the model already